docs/user_configuration_file.md

```diff
@@ -0,0 +1,53 @@
+# User Configuration File
+A file defined by users aimed at communicating specific user behavior is required to perform the various transaction processing steps inside of this application.
+
+The file format is YAML and its location can be passed as parameters to user requests requiring user-specific configuration. The YAML document fields are described below (also in YAML, so you get used to the user config format).
+
+```yaml
+InternalInternalTransferReferences:
+    - Description: References that appear in transfers between banks of the same user.
+      Type: List of Strings
+      Context: These references help detecting and removing internal transfers, which are no-op and add noise, therefore cleaning the transaction list.
+
+BankProcessingTimeInDays
+    - Description: The maximum amount of time, among all configured user banks, banks take to process transactions.
+      Type: Integer
+      Context: Related to "get_internal_transfer_references", it restricts the time window considered by the internal transfer cleaning algorithm.
+
+IncomeReferences
+    - Description: These are strings present in transactions references representing actual income.
+      Type: List of Strings
+      Context: Not all positive transactions are considered income, they can also be total or partial refunds. For a more realistic expense and income values, refunds are subtracted from expenses.
```

Comment:
Alright the text is missing some more clarification, but first let me try to get what you meant more specifically, what do you consider
1. "the purpose of the IncomeReferences"
2. "the context"
3. possible "misassumptions"

Some other confusion points that my explanation might have generated:
- When you say `incomes not defined in "IncomeReferences" (like refunds)`, incomes not defined in "IncomeReferences" are by definition not incomes, so I'm avoiding calling it incomes. I'm calling them "positive transactions not matching IncomeReferences".
- When you say `are not considered in the final report`, they (positive transactions not matching "IncomeReferences") are absolutely in the final report, they are just not considered as Income.
- When you say `applying the same rule to the related expenses as well.` , we don't have a "ExpenseReferences" configuration, so the rule only apply for selecting what is income, everything else currently falls into the expense automatically, maybe I need to clarify this better.

What about: `Context: Not all positive transactions are considered income, they can also be total or partial refunds. In order to avoid skewing income and expense numbers, positive transactions not matching strings configured in "IncomeReferences" are considered "expenses". In this way any refund, which is one case of a positive value not matching "IncomeReferences", automatically subtracts from "expenses".`

Reply:
As we talked about in the call. I was basing my thinking on how accounting usually uses these terms: income and expenses, where an income by concept cannot be an expense. Considering it, maybe an ordinary user could have the same misunderstanding as me. However, since the software is already working well and straightforwardly, any attempt to change this concept now is simply not worth it. Let's see if we face any troubles regarding that in the future, to decide better how to deal with that. As you suggested, it may be helpful to provide additional details about the software's behaviours in a manual.